mono/metadata/class.c

```diff
@@ -1565,17 +1565,33 @@ mono_class_interface_offset_with_variance (MonoClass *klass, MonoClass *itf, gbo
 	*non_exact_match = FALSE;
 	if (i >= 0)
 		return i;
-	
+
 	if (itf->is_array_special_interface && klass->rank < 2) {
 		MonoClass *gtd = mono_class_get_generic_type_definition (itf);
+		int found = -1;
+
+		for (i = 0; i < klass->interface_offsets_count; i++) {
+			if (mono_class_is_variant_compatible (itf, klass->interfaces_packed [i], FALSE)) {
+				found = i;
+				*non_exact_match = TRUE;
+			}
```

Comment:
Okay, I re-read the commit message and I guess I understand what's going on.

What's throwing me off is that the loop is essentially:
```c
found = -1;
for (i = 0; i < N; i++) {
  if (something (items[i])) {
    found = i;
  }
}
```
and I am getting hung up on what happens if the `something` returns TRUE for multiple `items[i]`s - we will then pick the last one.  I guess I expected a `break` in this loop (ie - pick the first one).

It seems like non-arrays will go down this path too - right -  that was the `klass->rank < 2` thing we noticed earlier. (If `klass->rank == 1` it's an array; if `klass->rank == 0` it's a non-array MonoClass.) And I guess I don't see why some generic MonoClass couldn't implement, say ``IEnumerable`1`` multiple ways - and why in that case we'd want the last one.